scipy/special/specfun.py

```diff
@@ -23,7 +22,6 @@
     'lamn',
     'lamv',
     'lpmn',
-    'lpn',
```

Comment:
I think you possibly want to leave `lpn` in. Since `lpn` is in `scipy.special` when you call `scipy.special.specfun.lpn` it will give you the one in `_basic.py` not `_specfun.pyx`

Reply:
The five functions here are the ones we should keep
![image](https://github.com/scipy/scipy/assets/60778417/963c0c4c-fedf-4abd-acfd-edeaf8d2d78b)

Reply:
Thanks @j-bowhay! This was very helpful, all fixed on this front.